### ISSUE TITLE
Elastic File System (EFS) - Setting up

### ISSUE THREAD

**Issue by leobardoramon** [gosha1128/NYUAMS#639]
I'm using the AWS "Getting Started" instructions to set up an EFS for a new QA Environment. See instructions here:
https://docs.aws.amazon.com/efs/latest/ug/getting-started.html

<img width="1744" alt="screen shot 2016-10-07 at 1 35 10 pm" src="https://cloud.githubusercontent.com/assets/2165796/19199658/0802d180-8c93-11e6-9118-e0016b5c4d76.png">

I'm having trouble mounting the EFS to the QA server , QAEBS_testing1

See my steps here:
https://docs.google.com/document/d/1ME7FYreEJvubhdojYUkpj73ThZEh3eFIW4ph5W9hcg4/edit

<img width="822" alt="screen shot 2016-10-07 at 1 24 51 pm" src="https://cloud.githubusercontent.com/assets/2165796/19199688/38b39396-8c93-11e6-9cd1-245c5ab5a071.png">

any suggestions?

**Comment by gosha1128** [comment]
Yes, "yum" is for red-hat and centos linux.

"apt-get" is the one to use for ubuntu.

Also, what error do you get when you run "mount" ?

-g

**Comment by leobardoramon** [comment]
Response:
mount.nfs4: Failed to resolve server us-east-1d.fs-eff338a6.efs.us-east-1d.amazonaws.com: Name or service not known

**Comment by gosha1128** [comment]
and why did you use that name?

server us-east-1d.fs-eff338a6.efs.us-east-1d.amazonaws.com

**Comment by leobardoramon** [comment]
I was trying to go by the instructions

See:
<img width="893" alt="screen shot 2016-10-07 at 1 54 47 pm" src="https://cloud.githubusercontent.com/assets/2165796/19200190/c244f4b8-8c95-11e6-9ac8-b15b2ddd847d.png">

I may be using the wrong format for "aws-region"

**Comment by gosha1128** [comment]
OK, do you know your AWS region ?

google "how do i get my aws region"

**Comment by gosha1128** [comment]
The following table lists the regions provided by an AWS account. Note that
you can't describe or access additional regions from an AWS account, such
as AWS GovCloud (US) or China (Beijing).
CodeName

us-east-1

US East (N. Virginia)

us-west-2

US West (Oregon)

us-west-1

US West (N. California)

eu-west-1

EU (Ireland)

eu-central-1

EU (Frankfurt)

ap-southeast-1

Asia Pacific (Singapore)

ap-northeast-1

Asia Pacific (Tokyo)

ap-southeast-2

Asia Pacific (Sydney)

ap-northeast-2

Asia Pacific (Seoul)

ap-south-1

Asia Pacific (Mumbai)

sa-east-1

South America (São Paulo)

**Comment by leobardoramon** [comment]
So, going by this diagram, the region would be "us-east-1"

<img width="886" alt="screen shot 2016-10-07 at 2 00 26 pm" src="https://cloud.githubusercontent.com/assets/2165796/19200343/86ec9d52-8c96-11e6-90d5-32c05e11482f.png">

I was providing the "Availability Zone"...

Should I try running the same command with the correct aws-region?

>>mount -t nfs4 -o nfsvers=4.1,rsize=1048576,wsize=1048576,hard,timeo=600,retrans=2 $(curl -s http://169.254.169.254/latest/meta-data/placement/availability-zone).fs-eff338a6.efs.us-east-1.amazonaws.com:/ efs

( I just removed the "d" from us-east-1 )

**Comment by gosha1128** [comment]
Yes.  Also, I think your region is listed in your account settings...

-g

**Comment by leobardoramon** [comment]
now I'm stuck here:

<img width="867" alt="screen shot 2016-10-07 at 2 43 21 pm" src="https://cloud.githubusercontent.com/assets/2165796/19201732/05e0ab3e-8c9d-11e6-94e3-debf8e9892d2.png">
<img width="426" alt="screen shot 2016-10-07 at 2 43 27 pm" src="https://cloud.githubusercontent.com/assets/2165796/19201731/05e03564-8c9d-11e6-8445-8badcde5bb47.png">

**Comment by leobardoramon** [comment]
I replaced "ec2-user" with "root" and it seemed to work fine with the rest of the instructions

**Comment by leobardoramon** [comment]
Everything looks good.

rsync is nearly complete as of now.

10/6 notes: https://docs.google.com/document/d/1ME7FYreEJvubhdojYUkpj73ThZEh3eFIW4ph5W9hcg4/edit

10/7 notes:
https://docs.google.com/document/d/1SuGvc1kwy9C0cdIJfpJs121NSVkdQmbsauWQ3TI9VlY/edit#

**Comment by leobardoramon** [comment]
Question:

Will there be some way to mount this File System to the LGS machines locally?

**Comment by leobardoramon** [comment]
Opened AWS Support Ticket:

Hello,

I've just finished syncing up data from QAFILES to the new EFS , QAEFS.

Now I am wondering if there is a way to mount QAEFS to my local machine so that I can access various files...

Before I would do this from my mac via Finder>"Connect to Server..." by entering the "Server Address" nfs://<file server IP address>/<file directory name> aka nfs://52.205.119.184/amsfiles

...I'm not really sure what the new Server Address or method would be now.

Any advice would be greatly appreciated.

Cheers,


Leobardo
leo@codetodesign.com

**Comment by leobardoramon** [comment]
Upon learning that this is not possible, Andrew has decided to shelf this idea.

Reply (Amazon):
![screen shot 2016-10-13 at 2 30 30 pm](https://cloud.githubusercontent.com/assets/2165796/19361711/c534c7aa-9151-11e6-99f2-cd9f5d719b24.png)

Closing

**Comment by gosha1128** [comment]
It is possible with that SSH idea.  But you have to try it to see...

George

**Comment by leobardoramon** [comment]
What ssh idea?

**Comment by gosha1128** [comment]
read the email carefully from amazon...you can setup a VPN connection
between the machines you want and the EFS backend...that should work !

-g
